lib/oppen/printer.rb

```diff
@@ -249,7 +262,23 @@ def advance_right
     def advance_left(token, token_width)
       return if token_width.negative?
 
-      print_stack.print token, token_width
+      nb_trims_on_break =
+        if token.is_a?(Token::Break)
+          # Find the first String token.
+          i = (left - 1) % scan_stack.length
+          i = (i - 1) % scan_stack.length while i != right && tokens[i] && !tokens[i].is_a?(Token::String)
+
+          # Count the number of consecutive Whitespace tokens.
+          res = 0
+          while i != right && tokens[i].is_a?(Token::Whitespace)
+            res += 1
+            i = (i - 1) % scan_stack.length
+          end
+
+          res
+        end
+
```

Comment:
This is an attempt to write it in a more "functional" (read idiomatic/readable/maintainable) way:

```ruby
trims_on_break = 
  if token.is_a?(Token::Break)
    # Find the first String token index
    idx = 
      (left - 1)
        .downto(right)
        .map { |i| i % scan_stack.length }
        .find { |i| tokens[i].is_a?(Token::String) }

    # Count consecutive Whitespace tokens
    if idx
      tokens[idx..right]
        .reverse
        .take_while { |t| t.is_a?(Token::Whitespace) }
        .size
    end
end
```

Please test it, I did it from memory :)

Reply:
This does not work. We have a circular array, so left can be greater or smaller than right. If left is smaller than right, `downto` will return an empty enum.